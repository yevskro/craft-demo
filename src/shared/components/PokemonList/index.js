import React from 'react';
import styled from 'styled-components';

function PokemonList({ children }) {
  return <StyledPokemonList>{children}</StyledPokemonList>;
}
const StyledPokemonList = styled.div`
  margin-top: 10px;
  text-align: center;

  @media (min-width: 400px) {
    /* tablet view */
    margin-left: 5%;
    margin-right: 5%;
  }

  @media (min-width: 700px) {
    /* desktop view */
    margin-left: 11%;
    margin-right: 11%;
  }
`;

export default PokemonList;
