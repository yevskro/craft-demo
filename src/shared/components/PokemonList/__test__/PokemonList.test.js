import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import PokemonList from '..';

test('renders without crashing', () => {
  render(<PokemonList />);
});

test('renders children', () => {
  render(<PokemonList>test</PokemonList>);
  expect(screen.getByText('test')).toBeInTheDocument();
});

test('snapshot', async () => {
  const tree = renderer.create(<PokemonList>test</PokemonList>).toJSON();
  expect(tree).toMatchSnapshot();
});
