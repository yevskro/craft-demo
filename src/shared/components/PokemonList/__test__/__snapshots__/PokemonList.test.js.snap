// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  margin-top: 10px;
  text-align: center;
}

@media (min-width:400px) {
  .c0 {
    margin-left: 5%;
    margin-right: 5%;
  }
}

@media (min-width:700px) {
  .c0 {
    margin-left: 11%;
    margin-right: 11%;
  }
}

<div
  className="c0"
>
  test
</div>
`;
